### DISCUSSION TITLE
How to learn webkit more efficiently, seek some advice, and some problems encountered in the study, can someone help me?

### DISCUSSION THREAD

**Discussion by Geroge-W** [Igalia/cog#527] [Q&A]
Hi, i am a beginer with the cog(wpewebkit) and i tried to count the total memory by some ways, shuch as the system /proc/pid/smaps, web inspector memory and set the Env WEBKIT_SAMPLE_MEMORY. But I can't understand what these memory represent, forexample , i open the url about:blank with cog, the memory are as follows: 
inspector: Maximum=40MB(JavaScript=6.94MB;Page=33.83MB)
smaps: sum of VMRSS = 102MB
Resident Set Bytes of WEBKIT_SAMPLE_MEMORY=102MB
the code of libWPEWebKit-1.0.so.3.17.10: text=36.5, rodata=1.9, data=0.2, bss=0.13,total=39MB
1. firstly, the memory of code + inspector is about  80MB，Where did the remaining about 20MB of memory go? and How to count the total memory at cog runtime accurately?
2. secondly, I want to use gdb to understand the code design framework, but found that it doesn't work, is there any recommended way to understand the code framework of WPEWebKit?
3. finally, i tried to make the SHARED library of libwpe, backend-fdo, wpewebkit and cog to STATIC library, then, Link all into one executable program, but it failed at runtime when it loading the gio-extension-module('wl' and 'drm'). Does WPEWebKit provide a way（macro） to package it into a static executable program? For a beginer, is it recommended to package these libraries into a static executable program? Will it be complicated?

Above all, Thanks!

**Comment by philn** [comment]
About 1. maybe @clopez can answer...
About 2. due to the multi-process nature of WPE, debugging is not trivial, you can find some hints there: https://trac.webkit.org/wiki/WebKitGTK/Debugging
About 3. AFAIK that was never attempted. Definitely not a beginner-level task.
